### ISSUE TITLE
Computed property names and arrow functions not supported

### ISSUE THREAD

**Issue by hackel** [Studio3T/robomongo#1252]
In Robomongo 0.9.0, the following query produces an error:
```js
db.categories.find({['na' + 'me']: 'test'})
// Error: Line 1: Unexpected token [
```
The exact same query works just fine in the MongoDB Shell, version 3.2.10.  I'm a bit confused by this.  `interpreterVersion()` shows `MozJS-38` in both RoboMongo and mongo cli.  Was the functionality added after 3.2.0-26-g203d04a?  According to [this](https://docs.mongodb.com/manual/release-notes/3.2-javascript/), it should already be there in 3.2.  Arrow functions also fail in Robomongo, but work in the cli client.

Edit: I guess this is really a dupe of #1197.

**Comment by simsekgokhan** [comment]
Hi @hackel , thanks for reporting the problem. We have reproduced the computed property names issue in our labs. We will investigate, and about arrow functions, yes that is the duplicate, unfortunately we didn't have a chance to investigate that one yet.

**Comment by simsekgokhan** [comment]
It seems like we have an issue related to parsing array brackets which needs to be investigated in the code.

Can you try the following workaround solution? It is not very user friendly but I hope it will help you at least for now. It worked for us.

```
var query = {};
var key = 'na' + 'me';
query[key] = 'test';
db.categories.find(query);

```

**Comment by hackel** [comment]
Yes, that works fine.  I believe this is related to all es6-specific features.  If you use "let" instead of "var", for example, it gives a similar error.

**Comment by simsekgokhan** [comment]
Hi @hackel ,
We did some investigation and seems like we identified the initial changes to support ES6 features where your original query works. Currently, ES6 support is planned to be in next release with 3.4 support.

However it is possible to provide a beta version if you do not want to wait and test the beta now.
Please share your OS if you are interested in this beta.

**Comment by hackel** [comment]
@simsekgokhan Awesome! I'm happy to test it out if that would help you guys out.  I haven't run into this issue since I first reported it, as it was kind-of a one-off.  I'm on Linux.

**Comment by simsekgokhan** [comment]
@hackel 
No worries, please see the following patch.
The patch is on top of Robomongo 1.0. Only change is support for "ECMAScript 2015 (Modernized JavaScript Implementation aka ES6)"
https://github.com/Studio3T/robomongo/releases/tag/v1.0.2

**Comment by simsekgokhan** [comment]
@hackel 
Even a better option, this beta includes both ECMAScript 2015 and 3.4 support, fyi:  [Robomongo 1.1 - Beta version with MongoDB 3.4 Support](https://github.com/Studio3T/robomongo/issues/1250#issuecomment-301036770)

**Comment by juliashibalko** [comment]
Fixed and verified in Robo3T 1.1 version. Closing as fixed.
Please feel free to reopen in any case.
